### PR TITLE
json imp. and count argument

### PR DESCRIPTION
**Pull request by karknu**
Spit out a propper json array of samples.
Add a `-c` argument which limits the number of pings to send

### CODE REVIEW

File: network-mux/demo/cardano-ping.hs [hunk 1]
```diff
@@ -20,7 +20,7 @@ import           Control.Tracer (nullTracer)
 import           Data.Aeson hiding (Options, json)
 import           Data.ByteString.Lazy (ByteString)
 import qualified Data.ByteString.Lazy as BL
-import qualified Data.ByteString.Lazy.Char8 as C8 (putStrLn)
+import qualified Data.ByteString.Lazy.Char8 as C8 (putStr)
```
Comment (coot) [inline network-mux/demo/cardano-ping.hs:23]: A more commonly used name is `BSC`.

File: network-mux/demo/cardano-ping.hs [hunk 2]
```diff
@@ -320,23 +326,26 @@ pingClient Options{quiet, magic, json} peer = bracket
            then return (msBlob sdu, t_e)
            else nextMsg bearer timeoutfn ptclNum
 
-    keepAlive :: MuxBearer IO -> TimeoutFn IO -> String -> TDigest 5 -> Word16 -> IO ()
+    keepAlive :: MuxBearer IO -> TimeoutFn IO -> String -> TDigest 5 -> Word32 -> IO ()
+    keepAlive _ _ _ _ cookie | cookie == maxCount = return ()
     keepAlive bearer timeoutfn peerStr td cookie = do
-        !t_s <- write bearer timeoutfn $ wrap keepaliveNum InitiatorDir (keepAliveReq cookie)
+        when (cookie > 0 && json) $ putStrLn ","
```
Comment (coot) [inline network-mux/demo/cardano-ping.hs:332]: This is not very elegant, but I don't have an idea how to make it nicer.

**Review by coot — APPROVED** [review]

File: network-mux/demo/cardano-ping.hs (hunk 1 above)
Comment (karknu) [reply network-mux/demo/cardano-ping.hs:23]: Will change.

File: network-mux/demo/cardano-ping.hs (hunk 2 above)
Comment (karknu) [reply network-mux/demo/cardano-ping.hs:332]: It also doesn't work when we're pinging multiple hosts. Will think of another way.

File: network-mux/demo/cardano-ping.hs [hunk 3]
```diff
@@ -269,8 +302,8 @@ toStatPoint ts host cookie sample td =
     stddev' = fromMaybe 0 (stddev td)
 
 
-pingClient :: Options -> AddrInfo -> IO ()
-pingClient Options{quiet, magic, json} peer = bracket
+pingClient :: (LogMsg -> IO ()) -> Options -> AddrInfo -> IO ()
```
Comment (coot) [inline network-mux/demo/cardano-ping.hs:305]: Why not just use `Tracer IO LogMsg`?

Comment (karknu) [reply network-mux/demo/cardano-ping.hs:305]: I have no confident in `Tracer`.

Comment (coot) [reply network-mux/demo/cardano-ping.hs:305]: You are using it!

Comment (karknu) [reply network-mux/demo/cardano-ping.hs:305]: Any similarity with `Tracer` was unintentional.

Comment (coot) [reply network-mux/demo/cardano-ping.hs:305]: `Tracer` was modelled exactly after this case and  `-> a` is **the cannonical** example of a contravariant functor.

Comment (karknu) [reply network-mux/demo/cardano-ping.hs:305]: I would never have seen it. Will make the adjustment.

**Pull request merged**